components/reusable-components/BlogCard/BlogCard.tsx

```diff
@@ -0,0 +1,37 @@
+import Link from 'next/link';
+import { HiArrowLongRight } from 'react-icons/hi2';
+
+import style from './blogCard.module.scss';
+
+interface BlogCardProps {
+  id: string;
+  title: string;
+  body: string;
+}
+
+interface TruncateProps {
+  str: string;
+  noWords: number;
+}
+
+function BlogCard({ id, title, body }: BlogCardProps) {
+  const truncate = ({ str, noWords }: TruncateProps) => {
+    return str.split(' ').splice(0, noWords).join(' ');
```

Comment:
Although the use of splice here won't cause any side effects, it is better to use slice since it does not modify the original array but instead returns a new array. (pure functions)